### ISSUE TITLE
Remove dependency to OIS

### ISSUE THREAD

**Issue by hwoarangmy** [OpenDungeons/OpenDungeons#1268]
@oyvindln I think you have started working on this point. Are you still working on it ?

**Comment by oyvindln** [comment]
I started on it yea, and committed some stuff, I haven't worked much on OD lately though.

**Comment by hwoarangmy** [comment]
wow, that's fast answering ^^
Are you still on it ? Do you want to assign this issue ? If you want, I can help as I should have some free time.

**Comment by hwoarangmy** [comment]
@oyvindln BTW, I've tried your PR and it seems I get some weird results with windows
![sfml_window](https://cloud.githubusercontent.com/assets/7493706/21505067/57b76180-cc64-11e6-8815-ef63962243ce.png)

Please tell me if you want me to test something

**Comment by oyvindln** [comment]
Hm, no idea why it would be like that. Does it happen if OD_USE_SFML_WINDOW is not set to ON(in cmake)?

**Comment by hwoarangmy** [comment]
If I don't set it, display is as it used to be. Note that I've tried after compiling with MinGW. I've not tried with MSVS2013. Have you ?

**Comment by oyvindln** [comment]
I used MSVS2015 when I was testing on windows. I can't remember having that issue, but I'll have a look.

**Comment by oyvindln** [comment]
I seem to have some path issue that makes some of the file paths contain a null character, I will have to try to solve that first before it's possible to test.

**Comment by hwoarangmy** [comment]
well, don't worry about that, i will test tomorrow. I have functional environments with MSVS2013 and MinGW (which are what we said we officially support). I will tell you if the issue is the same or not.

**Comment by hwoarangmy** [comment]
I've tried the MSVS2013 version. With OpenGL renderer, I get the same behaviour. With the D3D renderer, as expected, it doesn't work. I've not checked what happens in details but since we explicitely use the open gl renderer, I didn't expect it to work.
Note that the glitch also happens in the main screen so it is not related to ingame settings. It is like something is messed with the z buffer and meshes overlap.

**Comment by hwoarangmy** [comment]
@oyvindln I've had a quick look and it seems to work when creating the SFML window with a dept of 32:
```
sf::RenderWindow sfmlWindow(sf::VideoMode(w, h, sf::VideoMode::getDesktopMode().bitsPerPixel)
, windowTitle, style, sf::ContextSettings(32, 0, 2, 1));
```

Does it also work for you ?
Note that I've tried the minimap camera with the SFML window and it seems pretty laggy. It seems pretty less efficient than the OIS version (at least the debug version). I will try a release when I can.

**Comment by oyvindln** [comment]
Changing the depth bits to 32 seems to work fine for me on linux. (No idea what the ideal value is though.)
I haven't tested on windows yet as I haven't looked into fixing the path issue that is happening there. (For some reason the paths output when trying to load files have a null character in the middle.)

I guess whatever the depth buffer is set to if the parameter is 0 is making the depth buffer too small on windows.

**Comment by goupildb** [comment]
On my 32-bits Debian Sid, the compilation stop on:
[ 40%] Building CXX object CMakeFiles/opendungeons.dir/source/game/Skill.cpp.o
[ 41%] Building CXX object CMakeFiles/opendungeons.dir/source/game/SkillManager.cpp.o
[ 41%] Building CXX object CMakeFiles/opendungeons.dir/source/game/SkillType.cpp.o
In file included from /mnt/DDprc/Download/OpenDungeons-0.7.1/source/modes/GameEditorModeBase.h:21:0,
                 from /mnt/DDprc/Download/OpenDungeons-0.7.1/source/game/SkillManager.cpp:23:
/mnt/DDprc/Download/OpenDungeons-0.7.1/source/modes/AbstractApplicationMode.h:24:10: fatal error: OIS/OISKeyboard.h: Aucun fichier ou dossier de ce type
 #include <OIS/OISKeyboard.h>
          ^~~~~~~~~~~~~~~~~~~
compilation terminated.
CMakeFiles/opendungeons.dir/build.make:1271 : la recette pour la cible « CMakeFiles/opendungeons.dir/source/game/SkillManager.cpp.o » a échouée
make[2]: *** [CMakeFiles/opendungeons.dir/source/game/SkillManager.cpp.o] Erreur 1
make[2]: *** Attente des tâches non terminées....
CMakeFiles/Makefile2:67 : la recette pour la cible « CMakeFiles/opendungeons.dir/all » a échouée
make[1]: *** [CMakeFiles/opendungeons.dir/all] Erreur 2
Makefile:129 : la recette pour la cible « all » a échouée
make: *** [all] Erreur 2

libois-dev is installed

[CMakeError.log](https://github.com/OpenDungeons/OpenDungeons/files/1961109/CMakeError.log)
[CMakeOutput.log](https://github.com/OpenDungeons/OpenDungeons/files/1961111/CMakeOutput.log)

**Comment by akien-mga** [comment]
Debian puts its OIS headers in `/usr/include/ois` instead of the default upstream `/usr/include/OIS`, which we expect: http://metadata.ftp-master.debian.org/changelogs/main/o/ois/ois_1.3.0+dfsg0-6_changelog
There should be a `/usr/include/ois` symlink to `/usr/include/OIS` though, that should keep things working.

**Comment by goupildb** [comment]
Thank you very much, it works with:
# ln -s /usr/include/ois /usr/include/OIS

For information, it seems that the Debian maintainer has added in the meantime the symbolic link in the package.
The problem can therefore also be solved with a:
# apt install reinstall libois-dev

Thank you

**Comment by goupildb** [comment]
Sorry for the size of the font above :)

**Comment by akien-mga** [comment]
Indeed, I didn't think to ask that as that symlink was added in 2012 and I would expected it already present on your Debian Sid :)

**Comment by akien-mga** [comment]
Oops, reopening, the issue was initially about something else (please open a specific issue next time).

**Comment by goupildb** [comment]
2012? :))
I think I had installed the libois-dev library in February 2011 on the occasion of my [first test](http://download.tuxfamily.org/sdtraces/BottinHTML/Bottin_K-O_files/Open_Dungeons-12137.html) of your game (0.4.6): it seems coherent :)

Sorry, my compilation problems were related to OIS, I thought that putting it here was suitable, but you're right, I should have created another Issue
